### PR TITLE
Small code cleanup

### PR DESCRIPTION
**Pull request by AdrianLxM**
I just did some small changes to make the code better readable:

changed local variables to lower case (Usually class names start with capital letters, method calls are now distinguishable from static calls)
changed constructor to private (full singleton design pattern)
ran automatic code rearrangement in Android Studio
ran automatic code formatting standardization in Android Studio

### CODE REVIEW

**Comment by tzachi-dar** [comment]
Thanks for creating this change. It looks good.

It seems fine, however I'll be going abroad in a few days and I don't want
to make any change right now.
Once I'm back, I'll commit this.

Thanks
Tzachi

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 1]
```diff
@@ -294,117 +285,114 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
-    	} catch (InterruptedException e) {
-    	    Log.e(TAG, "cought InterruptedException! ", e);
+                if (lastReadingArr != null && lastReadingArr.length > 0) {
+                    // Last in the array is the most updated reading we have.
+                    TransmitterRawData lastReading = lastReadingArr[lastReadingArr.length - 1];
+
+                    //if (lastReading.CaptureDateTime > lastReportedReading + 5000) {
+                    // Make sure we do not report packets from the far future...
+                    if ((lastReading.CaptureDateTime > lastReportedTime) &&
+                            (!almostEquals(lastReading, lastReportedReading)) &&
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [Useless parentheses.](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l328)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 2]
```diff
@@ -294,117 +285,114 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
+                break;
+            }
         }
-		Log.e(TAG, "EXITING mstop=true" );
+        Log.e(TAG, "EXITING mstop=true");
     }
 
-    public void Stop()
-    {
+    public void Stop() {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [Method names should not start with capital letters](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l390)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 3]
```diff
@@ -294,117 +285,114 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
-	        }
-    	} catch (InterruptedException e) {
-    	    Log.e(TAG, "cought InterruptedException! ", e);
+                if (lastReadingArr != null && lastReadingArr.length > 0) {
+                    // Last in the array is the most updated reading we have.
+                    TransmitterRawData lastReading = lastReadingArr[lastReadingArr.length - 1];
+
+                    //if (lastReading.CaptureDateTime > lastReportedReading + 5000) {
+                    // Make sure we do not report packets from the far future...
+                    if ((lastReading.CaptureDateTime > lastReportedTime) &&
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [Useless parentheses.](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l327)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 4]
```diff
@@ -175,55 +170,53 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
-    	if(indexOfSlash == -1) {
-    		// We can not find a collection name
-    		Log.e(TAG, "Error bad dburi. Did not find a collection name starting with / " + dbury);
-    		// in order for the user to understand that there is a problem, we return null
-    		return null;
-    	}
-
-    	MongoWrapper mt = new MongoWrapper(dbury, collection, "CaptureDateTime", "MachineNameNotUsed");
-    	return mt.ReadFromMongo(numberOfRecords);
+    public static List<TransmitterRawData> readFromMongo(String dbury, int numberOfRecords) {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [Avoid reassigning parameters such as 'dbury'](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l173)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 5]
```diff
@@ -175,55 +170,53 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
+        MongoWrapper mt = new MongoWrapper(dbury, collection, "CaptureDateTime", "MachineNameNotUsed");
+        return mt.ReadFromMongo(numberOfRecords);
     }
 
     // format of string is ip1:port1,ip2:port2;
-    public static TransmitterRawData[] Read(String hostsNames, int numberOfRecords)
-    {
-        String []hosts = hostsNames.split(",");
-        if(hosts.length == 0) {
+    public static TransmitterRawData[] read(String hostsNames, int numberOfRecords) {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [The method read() has an NPath complexity of 56](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l202)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 6]
```diff
@@ -294,117 +285,114 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
-    {
-    	Long LastReportedTime = new Date().getTime();
-    	TransmitterRawData LastReportedReading = null;
-    	Log.e(TAG, "Starting... LastReportedReading " + LastReportedReading);
-    	try {
-	        while (!mStop && !interrupted()) {
-	        	// try to read one object...
-                TransmitterRawData[] LastReadingArr = null;
-                if(WixelReader.IsConfigured(mContext)) {
+    public void run() {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [The method run() has an NPath complexity of 15](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l308)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 7]
```diff
@@ -70,34 +69,32 @@ public static void sStop() {
...
         }
         return false;
     }
 
- // last in the array, is first in time
-    public static List<TransmitterRawData> Merge2Lists(List<TransmitterRawData> list1 , List<TransmitterRawData> list2)
-    {
-        List<TransmitterRawData> merged = new LinkedList <TransmitterRawData>();
+    // last in the array, is first in time
+    public static List<TransmitterRawData> merge2Lists(List<TransmitterRawData> list1, List<TransmitterRawData> list2) {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [The method merge2Lists() has an NPath complexity of 49](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l94)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java [hunk 8]
```diff
@@ -123,24 +120,22 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
...
+            mergedList = merge2Lists(mergedList, it);
         }
 
-        return MergedList;
+        return mergedList;
     }
 
-    public static List<TransmitterRawData> ReadHost(String hostAndIp, int numberOfRecords)
-    {
+    public static List<TransmitterRawData> readHost(String hostAndIp, int numberOfRecords) {
```
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java]: Codacy Issue found: [The method readHost() has an NPath complexity of 24](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l133)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 1 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:328]: Codacy Issue found: [Useless parentheses.](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l328)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 2 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:390]: Codacy Issue found: [Method names should not start with capital letters](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l390)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 3 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:327]: Codacy Issue found: [Useless parentheses.](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l327)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 4 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:173]: Codacy Issue found: [Avoid reassigning parameters such as 'dbury'](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l173)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 5 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:202]: Codacy Issue found: [The method read() has an NPath complexity of 56](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l202)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 6 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:308]: Codacy Issue found: [The method run() has an NPath complexity of 15](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l308)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 7 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:94]: Codacy Issue found: [The method merge2Lists() has an NPath complexity of 49](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l94)

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java (hunk 8 above)
Comment (mgranberry) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java:133]: Codacy Issue found: [The method readHost() has an NPath complexity of 24](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1313443989/issues/source?bid=2735053&fileBranchId=2735067#l133)

**Pull request closed without merging**